File: backend/models/job.py
from typing import Optional
from pydantic import BaseModel


class JobCreate(BaseModel):
    company_name: str
    role_name: str
    job_description: str
    apply_link: str
    h1b_sponsorship: bool
    salary: Optional[str] = None
    location: Optional[str] = None
    remote_type: Optional[str] = None
    job_type: Optional[str] = None
    experience_level: Optional[str] = None
    bookmarked: bool = False
    status: str = "new"
    date_posted: Optional[str] = None
    date_applied: Optional[str] = None
    resume_path: Optional[str] = None
    notes: Optional[str] = None
    recruiter_name: Optional[str] = None
    recruiter_linkedin: Optional[str] = None
    company_website: Optional[str] = None
    source: Optional[str] = None
    external_job_id: Optional[str] = None
    skills: Optional[str] = None
    department: Optional[str] = None
    benefits: Optional[str] = None


class JobUpdate(BaseModel):
    bookmarked: Optional[bool] = None
    status: Optional[str] = None
    notes: Optional[str] = None
    date_applied: Optional[str] = None
    resume_path: Optional[str] = None

File: frontend/src/types/job.ts
export interface Job {
  id: number
  company_name: string
  role_name: string
  job_description: string
  apply_link: string
  h1b_sponsorship: boolean
  salary: string | null
  location: string | null
  remote_type: string | null
  job_type: string | null
  experience_level: string | null
  bookmarked: boolean
  status: string
  date_posted: string | null
  date_applied: string | null
  resume_path: string | null
  notes: string | null
  recruiter_name: string | null
  recruiter_linkedin: string | null
  company_website: string | null
  source: string | null
  external_job_id: string | null
  skills: string | null
  department: string | null
  benefits: string | null
  created_at: string
}

export interface JobStats {
  total: number
  by_status: Record<string, number>
  by_source: Record<string, number>
  applied_this_week: number
  bookmarked: number
}

export type JobCreate = Omit<Job, 'id' | 'created_at'>
export type JobUpdate = Partial<Pick<Job, 'bookmarked' | 'status' | 'notes' | 'date_applied' | 'resume_path'>>
